File: LV 1/소수 만들기.py
def getodd():
    list = [True] * 3001
    list[1] = False
    list[2] = True
    for i in range(2, 3001):
        if list[i] == False:
            continue
        
        for j in range(i+i, 3001, i):
            list[j] = False
    
    return list

def solution(nums):
    list = getodd()
    answer = 0
    for i in range(len(nums)):
        for j in range(i+1, len(nums)):
            for k in range(j+1, len(nums)):
                if i == j or i == k or j == k:
                    continue
                s = nums[i] + nums[j] + nums[k]
                if list[s] == True:
                    answer += 1
    return answer

File: LV 1/소수 만들기.java
class Solution {
    public boolean determinePrimeNumber(int number) {
        if (number == 1) {
            return false;
        }
        if (number == 2) {
            return true;
        }
        for (int i = 2; i<number; i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    public int solution(int[] nums) {
        int answer = 0;
        for (int i = 0; i < nums.length; i++) {
            for (int j = i+1; j < nums.length; j++) {
                for (int k = j+1; k < nums.length; k++) {
                    if (determinePrimeNumber(nums[i] + nums[j] + nums[k])) {
                        answer += 1;
                    }
                }
            }
        }

        return answer;
    }
}
